src/lib/modes/mode_pad/mode_pad.h

```diff
@@ -63,16 +76,36 @@ class BOTAN_TEST_API BlockCipherModePaddingMethod {
       * virtual destructor
       */
       virtual ~BlockCipherModePaddingMethod() = default;
+
+   protected:
+      /**
+      * Applies the concrete padding to the @p last_block assuming the padding
+      * bytes should start at @p padding_start_pos within the last block.
+      *
+      * Concrete implementations of this function must ensure not to leak
+      * @p padding_start_pos via side channels.
+      */
+      virtual void apply_padding(std::span<uint8_t> last_block, size_t padding_start_pos) const {
+         BOTAN_UNUSED(last_block, padding_start_pos);
+         BOTAN_ASSERT_UNREACHABLE();
+      }
+
+      /**
+      * Removes the padding from @p last_block and returns the number of data
+      * bytes. If the padding is invalid, this returns the byte length of
+      * @p last_block.
```

Comment:
Likewise this should note that the input is poisoned and the implementation must be constant time